pkg/router/controller/factory/factory.go

```diff
@@ -124,6 +127,21 @@ func (factory *RouterControllerFactory) Create(plugin router.Plugin, watchNodes,
 		}, &kapi.Secret{}, secretEventQueue, factory.ResyncInterval).Run()
 	}
 
+	// Check ARP Cache and endpoint list
+	go utilwait.Forever(func() {
+		data, err := ioutil.ReadFile("/proc/sys/net/ipv4/neigh/default/gc_thresh3")
+		if err != nil {
+			glog.Warning("Error reading ARP neighbour information")
+		}
+		endpoints, _ := factory.KClient.Endpoints(factory.Namespace).List(metav1.ListOptions{})
+		items := len(endpoints.Items)
```

Comment:
Rather than running this every 10 minutes, could you just tie into the existing endpoints watch(es) and keep count of the active number of endpoints from there?